public/app/features/alerting/unified/api/featureDiscoveryApi.ts

```diff
@@ -63,6 +64,7 @@ export const featureDiscoveryApi = alertingApi.injectEndpoints({
         const rulerConfig = features.features.rulerApiEnabled
           ? ({
               dataSourceName: dataSourceSettings.name,
+              dataSourceUid: dataSourceSettings.uid,
```

Comment:
The `rulerConfig` object predates UID-based identifiers for data sources. Let's try to slowly migrate to UIDs